### ISSUE TITLE
collection.format hits maximum recursion depth for collections with lots of holes.

### ISSUE THREAD

**Issue by chrisburel** [4degrees/clique#24]
The following code gives an example.

```python
paths = ["name.{0:04d}.jpg".format(x) for x in range(2000)[::2]]
collection = clique.assemble(paths)[0][0]
collection.format("{head}####{tail}")
```

**Comment by martinpengellyphillips** [comment]
Thanks for the report. Will take a look.

**Comment by martinpengellyphillips** [comment]
Should be fixed now in new release 1.3.1.
